### ISSUE TITLE
ProxySockets have no 'pipe' function

### ISSUE THREAD

**Issue by Miserlou** [krisives/proxysocket#4]
It looks like ProxySockets can do many of the things that net.Sockets can, but not 'pipe'. Since many applications use streams and pipes now, this means that ProxySocket is less functional as a drop-in replacement now.

**Comment by krisives** [comment]
Well I want it to be easy to use and mimic as close as possible the Socket interface, so it should do this if possible. The easiest attempt at adding this support would be to simply call the original pipe() method, and either throw or buffer the attempts to call pipe() before the SOCKS5 negotiation is complete. Once the SOCKS5 negotiation is complete the Socket packets just flow as normal.

Try adding a chunk of code around here:

https://github.com/krisives/proxysocket/blob/cf903d7bb488b0ec12eb57bc5dc92592f8bbc1b4/proxysocket.js#L137

Something like this:

```
self.pipe = function (dest, opts) {
  return socket.pipe(dest, opts);
};
```

**Comment by Miserlou** [comment]
Fixed by PR #6
